_drafts/GDB Tips and Tricks.md

```diff
@@ -0,0 +1,500 @@
+---
+title: "Debugging with GDB: Tips & Tricks (nRF52 use case)"
+author: mafaneh
+---
+
+<!-- excerpt start -->
+If I had to choose one significant aspect that I was not aware of before starting my career as a firmware developer, it would be how much time is spent **not actually developing**, and instead **debugging** firmware!
+
+Not convinced?
+
+I'll leave you with a few figures taken from the 2017 Embedded/EETimes **Embedded Markets Study** survey which showcase the significance of debugging in the professional life of a firmware developer.
+<!-- excerpt end -->
+
+![](Graphics/Job-Functions.png)
+<br>
+
+![](Graphics/Design-time.png)
+<br>
+
+![](Graphics/Tool-improvement.png)
+<br>
+
+Let's face it..
+
+Debugging is **generally difficult**, and it gets even more difficult for firmware applications and has more limitations compared with a mobile or web application.
+
+This is due to a few factors, some of which are:
+
+- Limited availability of resources for debugging. For example, some bugs such as memory corruption are very difficult to diagnose using software breakpoints. In those scenarios, you would rely on hardware breakpoints and usually there is a limited number of hardware breakpoints for a specific architecture. This makes debugging complex bugs a challenging task.
```

Comment:
wouldn't you use watchpoints rather than breakpoints for memory corruption?

Reply:
Yes. I'll update to reflect this.